src/mpi/mpi_caf.c

```diff
@@ -3255,22 +3290,29 @@ get_MPI_datatype (gfc_descriptor_t *desc)
    GFC_DTYPE_TYPE_SIZE == GFC_TYPE_CHARACTER + 64*strlen
 */
   if ( (GFC_DTYPE_TYPE_SIZE(desc)-GFC_DTYPE_CHARACTER)%64==0 )
-      return MPI_CHARACTER;
+    {
+      MPI_Datatype string;
+      if (char_len == 0)
+	char_len = GFC_DESCRIPTOR_SIZE (desc);
+      MPI_Type_contiguous(char_len, MPI_CHARACTER, &string);
+      MPI_Type_commit(&string);
+      return string;
+    }
 
   caf_runtime_error ("Unsupported data type in collective: %ld\n",GFC_DTYPE_TYPE_SIZE (desc));
   return 0;
 }
 
 
 static void
-co_reduce_1 (MPI_Op op, gfc_descriptor_t *source, int result_image, int *stat,
-             char *errmsg, int src_len __attribute__ ((unused)), int errmsg_len)
+internal_co_reduce (MPI_Op op, gfc_descriptor_t *source, int result_image, int *stat,
+	     char *errmsg, int src_len, int errmsg_len)
```

Comment:
Is it possible to set your editor to use spaces and not tab characters? Also, it would be great to revert the leading indentation whitespace changes, or at least fixup white space in a separate commit from when substantive edits are made.